tc-server/src/main/java/com/tc/objectserver/handler/ProcessTransactionHandler.java

```diff
@@ -388,7 +388,8 @@ private void addMessage(ClientID sourceNodeID, EntityDescriptor descriptor, Serv
       capture.setTransactionOrderPersistenceFuture(transactionOrderPersistenceFuture);
       try {
         EntityID entityID = descriptor.getEntityID();
-        ManagedEntity temp = entityManager.createEntity(entityID, descriptor.getClientSideVersion(), consumerID, !capture.getNodeID().isNull());
+        boolean canDelete = !entityManager.getEntityLoader().isPermanentEntity(entityID);
+        ManagedEntity temp = entityManager.createEntity(entityID, descriptor.getClientSideVersion(), consumerID, canDelete);
```

Comment:
Given that the `EntityManager` is going to call the `EntityServerService` to create the entity, wouldn't it be easier to just remove the boolean parameter entirely and let the EntityManager implementation ask the server service either via a new method, or by checking for the annotations whether a given construction should be permanent or not?

Reply:
Sounds good.  What is our policy about changing interface method signatures (to remove the boolean)?

Reply:
interface face changes are fine as long as they are contained in tc-core.

Reply:
one thing to consider that while permanent and temporary entities are an annotation driven concept, they didn't start that way.  Right now, temporary and permanent entity concepts are not consistent with each other.  Permanent entities have a runtime aspect to them while temporary entities are annotation only.  We probably should decide which way to go and make them consistent with each other.

Reply:
> 
> 
> Given that the `EntityManager` is going to call the `EntityServerService` to create the entity, wouldn't it be easier to just remove the boolean parameter entirely and let the `EntityManager` implementation ask the server service either via a new method, or by checking for the annotations whether a given construction should be permanent or not?

done